molecule/default/playbook.yml

```diff
@@ -91,5 +113,48 @@
       args:
         port: 22
         state: started
-        delay: 300
+        delay: 240
         timeout: 600
+    - import_tasks: get_server_uptime.yml
+
+- name: Register utility container
+  hosts: infra1
+  tasks:
+    - shell: |
+        lxc-ls -1 | grep utility | head -n 1
+      register: utility_container
+
+- name: Check for servers after rebooting all
+  hosts: infra1
+  tasks:
+    - shell: |
+        lxc-attach -n "{{ utility_container.stdout }}"  \
+        -- bash -c '. /root/openrc ; \
+        openstack server list -c ID -f value'
+      register: server_instances
+      changed_when: false
+      failed_when: false
+
+- name: Delete all previously existing servers
+  hosts: infra1
+  tasks:
+    - shell: |
+        lxc-attach -n "{{ utility_container.stdout }}"  \
+        -- bash -c '. /root/openrc ; \
+        openstack server delete "{{ item }}"'
+      when:
+        - server_instances.rc == 0
+        - server_instances.stdout != ""
+      with_items: "{{ server_instances.stdout_lines }}"
+
+- name: Check for servers before entering Nova molecule tests
+  hosts: infra1
+  tasks:
+    - shell: |
```

Comment:
This task should also be gated on `- server_instances.stdout != ""`, because if no servers were found the first time, then there is no reason to check again.